### ISSUE TITLE
Forge installer doesnt get replaced with actual forge server jar

### ISSUE THREAD

**Issue by Osiris-Team** [HSGamer/MCServerUpdater#6]
I just tested it on my system and a popup opens up where I need to select server and then it installs it into the .minecraft folder.
And this repeats every time you enter .start to launch the server. 
Bild
looking into this issue
Arman Kandel (Osiris) — 28.12.2023 20:23
for now disable the server updater and give the absolute path in the start command which should be something like (on windows)
java -jar C:\Users\arman\AppData\Roaming\.minecraft\libraries\net\minecraftforge\forge\1.18.2-40.2.14\forge-1.18.2-40.2.14-server.jar 
Arman Kandel (Osiris) — 28.12.2023 20:27

**Comment by HSGamer** [comment]
ForgeUpdater has been removed because of this issue.
Will need to find a better & reliable way to fetch the installer jar and build server jar from that.

**Comment by Osiris-Team** [comment]
https://github.com/HSGamer/MCServerUpdater/issues/6
We could get the real server jar from this path maybe?

**Comment by Osiris-Team** [comment]
@HSGamer yeah just tested it out and it works, the command looks something like this, after downloading the installer:
`java -jar .\forge-1.20.2-48.1.0-installer.jar --installServer`

I'm still not sure where the server jar finally is, probably it doesnt exist and all the libs get added to the default mc server jar at boot, thats why autoplug should just update the start command to `start-command: ./run.sh` (or .bat if windows).

So can we add the original code for the forge updater back? And also add update the command to run the installer?

![image](https://github.com/HSGamer/MCServerUpdater/assets/59899645/feab6616-beda-47f9-9c67-97d9c251c04e)

**Comment by HSGamer** [comment]
> @HSGamer yeah just tested it out and it works, the command looks something like this, after downloading the installer:
> `java -jar .\forge-1.20.2-48.1.0-installer.jar --installServer`
> 
> I'm still not sure where the server jar finally is, probably it doesnt exist and all the libs get added to the default mc server jar at boot, thats why autoplug should just update the start command to `start-command: ./run.sh` (or .bat if windows).
> 
> So can we add the original code for the forge updater back? And also  update the command to run the installer?
> 
> ![image](https://github.com/HSGamer/MCServerUpdater/assets/59899645/feab6616-beda-47f9-9c67-97d9c251c04e)
> 

The problem is that it creates the full server setup, not a single server jar.

**Comment by Osiris-Team** [comment]
@HSGamer here is what ChatGPT says about the forge start command:

It looks like you've provided a command line configuration for launching a Minecraft Forge server using the Forge Mod Loader (FML) and some specific libraries. Let me break down the important components of this command:

1. **Classpath (specified by `-p`):**
   - Various JAR files are specified in the classpath, including libraries related to Google Guava, Forge, ASM (Objectweb ASM), secure modules, and other dependencies required for running a Forge server.

2. **Module Path (specified by `--add-modules ALL-MODULE-PATH`):**
   - This option indicates that all JAR files in the specified classpath will be treated as modules. This is a way to use the module system introduced in Java 9.

3. **Exports (specified by `--add-exports jdk.naming.dns/com.sun.jndi.dns=java.naming`):**
   - This option adds a specific export statement for the given package. In this case, it exports the `com.sun.jndi.dns` package to `java.naming`.

4. **System Properties (specified by `-D`):**
   - Several system properties are set using the `-D` flag. Notable ones include:
     - `java.net.preferIPv6Addresses=system`: This sets the preference for IPv6 addresses.
     - `ignoreList`: Specifies a list of JAR files to be ignored.
     - `libraryDirectory`: Specifies the directory where libraries are located.
     - `legacyClassPath`: Specifies the classpath for legacy dependencies.

5. **Main Class and Launch Parameters:**
   - The main class is specified as `net.minecraftforge.bootstrap.BootstrapLauncher`.
   - Additional parameters specific to Forge server launch are provided, such as `--launchTarget`, `--fml.forgeVersion`, `--fml.mcVersion`, `--fml.forgeGroup`, and `--fml.mcpVersion`.

It seems like this command is designed to launch a Minecraft Forge server with specific versions and configurations. Ensure that all the specified libraries are available in the specified directory, and the required dependencies are correctly set up. Additionally, make sure you have the appropriate Java version installed to run this Minecraft Forge server.

**Comment by Osiris-Team** [comment]
@HSGamer basically you can think of it as an extracted server jar. When usign the installer agin those files get updated so I dont see the problem.

**Comment by HSGamer** [comment]
> @HSGamer basically you can think of it as an extracted server jar. When usign the installer agin those files get updated so I dont see the problem.
> 

Another problem is the start script file. It's different for each OS.

**Comment by Osiris-Team** [comment]
The actual command inside the .sh and .bat should be the same. But this shouldn't be a worry of the server updater.

**Comment by HSGamer** [comment]
> The actual command inside the .sh and .bat should be the same. But this shouldn't be a worry of the server updater.

There are still problems with this updater
- If we want to provide a single server jar as the output, this can't do.
- If we provide the update for the installer, the user will need to run the `installServer` command AND the `start` file.
- If we want to support fetching old versions, the link to that is different from the one in newer versions (probably from 1.12 and above).

Since I don't work on Forge Updater, I will leave this to you or someone who can solve these problems above.

**Comment by Osiris-Team** [comment]
Instead of providing the jar as output we can provide the run.sh/bat script. This should be ok since its only for forge and there isn't really another alternative.

The user won't have to run anything, server-updater will run the installer jar.

Supporting older versions can be done later, its not a priority now I would say. Since those older versions probably stopped receiving updates anyways.

**Comment by HSGamer** [comment]
> The user won't have to run anything, server-updater will run the installer jar.
> 

We won't. It's different from Bungee Updater.

**Comment by Osiris-Team** [comment]
Im not following, why can't we run the installer jar?

**Comment by HSGamer** [comment]
> Im not following, why can't we run the installer jar?
> 

I mistyped. It's different from the Spigot Updater. 
The output of BuildTools is a single jar file so we can handle it easily as the output of the Server Updater.
 But the output of Forge Installer is a full server setup with several jar files. We can't handle them as a single output of the Server Updater.

**Comment by Osiris-Team** [comment]
Mhmm, meaning to calculate the checksum to see if there is an update available?
In that case we can return the installer jar instead for this to work?

**Comment by HSGamer** [comment]
> Mhmm, meaning to calculate the checksum to see if there is an update available?
> In that case we can return the installer jar instead for this to work?

Then there is the second problem I mentioned above. The user needs to run both installServer and the start script.
